Source/DafnyCore/PrecedenceLinter.cs

```diff
@@ -92,6 +92,8 @@ class PrecedenceLinterVisitor : TopDownVisitor<LeftMargin> {
           // a SeqSelectExpr is the open-bracket). To avoid looking at that internal structure, we instead
           // use the .StartToken for these expressions.
           column = expr.StartToken.col;
+        } else if (expr is BinaryExpr binaryExpr && binaryExpr.PrefixOp is not null) {
```

Comment:
Are you not able to do something like `binaryExpr.E0.RangeToken.EndToken.Next` so you don't need `binaryExpr.PrefixOp` and all the related changes?

Reply:
I think that you're right that this entire branch of the condition isn't necessary.

The `prefixOp` parameter is still necessary, I think, to ensure that it can go into the `FormatTokens` property (which is used to compute the `StartToken` property used in the previous branch).

Reply:
Then this comment applies to where FormatTokens is updated 😄